src/track/serato/beatgrid.h

```diff
@@ -122,6 +128,10 @@ class SeratoBeatGrid final {
         m_pTerminalMarker = pTerminalMarker;
     }
 
+    void setBeats(BeatsPointer pBeats,
```

Comment:
No need to pass a pass a pointer if you don't need to store this reference outside the scope of this function invocation. Simply pass `const Beats&`.

Reply:
References are not null able. Is it impossible that `m_pBeats` is null? I thought so. But we can't skip the function call in that case, because the exported beatgrid should be empty,  too.

Reply:
But then why not provide a separate function for creating an empty beat grid or omitting it entirely (depending on the Serato format)? Beat grid conversion should not need to care about null pointers from an outer context. If you extract this conditional behavior in a separate function this would improve readability and narrows the scope of each function.

Just a suggestion to foster functional decomposition. You decide what is appropriate here.

Reply:
The issue is that in this case I think it would lead to code duplication, because even if the m_pBeats is not null, it could still contain no beats. So I'd need to check that case inse the `setBeats` method anyway.